lib/portage/manifest.py

```diff
@@ -235,32 +238,21 @@ def _readManifest(self, file_path, myhashdict=None, **kwargs):
             else:
                 raise
 
-    def _read(self):
-        """Parse Manifest file for this instance"""
-        try:
-            self._readManifest(self.getFullname(), myhashdict=self.fhashdict)
-        except FileNotFound:
-            pass
-
     def _parseManifestLines(self, mylines):
         """Parse manifest lines and return a list of manifest entries."""
         for myline in mylines:
-            myentry = None
             for parser in self.parsers:
                 myentry = parser(myline)
-                if myentry is not None:
+                if myentry:
                     yield myentry
                     break  # go to the next line
 
-    def _parseDigests(self, mylines, myhashdict=None, mytype=None):
+    def _parseDigests(self, mylines, myhashdict={}, mytype=None):
```

Comment:
You definitely do not want to do this. See https://stackoverflow.com/questions/26320899/why-is-the-empty-dictionary-a-dangerous-default-value-in-python

Reply:
I've been bitten by this a fair few times!